[EA Forum only] digest ad in the post page footer

s-cheng opened the pull request:
Most of this PR is just moving the homepage RHS digest ad to a separate file. This way I could reuse it in the post page footer. Based on [this comment thread](https://docs.google.com/document/d/1BmbGKMlkFAILwoc9298kEAM5DW2AGKb1ygu9z0vGQ6I/edit?disco=AAAA9u9CuSE).

<img width="674" alt="Screen Shot 2024-01-10 at 9 31 26 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/92e13a51-e17b-4a0c-87b5-24078cd91a79">

-----
Maybe it looks better with a max width:

<img width="674" alt="Screen Shot 2024-01-10 at 9 36 38 PM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/26890e95-2a9f-46ad-a1cc-885a2272c970">

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201302964208280/1206324629482403) by [Unito](https://www.unito.io)

s-cheng commented on packages/lesswrong/components/ea-forum/DigestAd.tsx:
Most of this is the same with updated class names

s-cheng commented on packages/lesswrong/components/ea-forum/DigestAd.tsx:
```diff
@@ -0,0 +1,256 @@
+  const { captureEvent } = useTracking()
+  
+  // Only show this on sites that have a digest
+  if (!hasDigests) return null
```
Added this check

s-cheng commented on packages/lesswrong/components/ea-forum/DigestAd.tsx:
```diff
@@ -0,0 +1,256 @@
+  }
+  
+  return <AnalyticsContext pageSubSectionContext="digestAd">
+    <div className={classNames(classes.root, className)}>
```
Cleaned up the html a bit

s-cheng commented on packages/lesswrong/components/recommendations/PostBottomRecommendations.tsx:
```diff
@@ -13,6 +14,13 @@ const styles = (theme: ThemeType) => ({
+      paddingRight: 8,
+      marginLeft: -8,
+      marginRight: -8,
+    },
```
Also fixed this

agnesstenlund commented:
screenshots look great! I don't see it on localhost though, do I need to do something else?

agnesstenlund commented:
(Not sure yet which width I prefer, need to see it more in context)

s-cheng commented:
@agnesstenlund If you dismiss the ad in one place (like the home RHS) then it's dismissed everywhere, so prob that's why it's not appearing? You can try viewing a post page via incognito to see it

Will-Howard reviewed (approved):
The code looks good 👍 , just a couple of UI things:

1. This button goes slightly higher than the input. I'm not sure if this is intentional but the fact that there is only a tiny difference makes it look off to me
<img width="456" alt="Screenshot 2024-01-12 at 10 56 57" src="https://github.com/ForumMagnum/ForumMagnum/assets/77623106/67265e07-5386-4d2f-a418-c0ef16610183">

2. On mobile the fonts look a bit to small, especially compare to the section below
<img width="456" alt="Screenshot 2024-01-12 at 10 59 44" src="https://github.com/ForumMagnum/ForumMagnum/assets/77623106/15e29a30-5deb-4c4a-bb4b-06a713d51b02">
3. I think the padding above the ad is a bit too much (it's the same as it was before with just the recommendations, but I think because the ad is smaller it seems like an unusually large gap)

agnesstenlund commented:
agree with Will it'd be nice if the input was also 40px tall (like the arrow button)

I also tried changing the font to match the other fonts on the bottom of the page, but that also changed the RHS styling, so probably not worth creating a separate thing. But I pushed some padding changes.
![Screenshot 2024-01-16 at 14 09 53](https://github.com/ForumMagnum/ForumMagnum/assets/13235378/88bc4d54-5dc6-483b-b483-6125382181c0)
![Screenshot 2024-01-16 at 14 13 38](https://github.com/ForumMagnum/ForumMagnum/assets/13235378/12102ed7-4026-4850-8319-b87ae504e3aa)

Another thing I'd still really like is getting rid of the little * after Email address (which I don't think is a standard pattern, and it should be fairly obvious it's a mandatory input?), as well as making the text black when it's pre-filled for users. I still think the grey text looks so much like a placeholder, I think I'd be confused if I saw my email in there. (Ideally the black text should also be editable so you can use another email, but not a big deal if you can't)

agnesstenlund commented:
actually, I did a hacky thing where the font sizes will be bigger on medium to smaller screens (since those screens don't have a RHS)

s-cheng commented:
Ok I've replaced the MUI input with a plain one to remove the * and keep the text black even when it's disabled, and updated the fonts on the post page footer version. Also changed the RHS version to just use the arrow button since the "Subscribe" version was taking up a lot of space.

<img width="265" alt="Screen Shot 2024-01-16 at 11 44 59 AM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/9da16d4e-cd94-45ef-8785-5c7ec24c080c">

<img width="689" alt="Screen Shot 2024-01-16 at 11 48 57 AM" src="https://github.com/ForumMagnum/ForumMagnum/assets/9057804/fc0260b7-cc90-4690-996d-1da3a6b2a12c">

agnesstenlund commented:
Perfect 🥹
